sdks/python/apache_beam/pipeline.py

```diff
@@ -172,6 +172,10 @@ def __init__(self, runner=None, options=None, argv=None):
     # If a transform is applied and the full label is already in the set
     # then the transform will have to be cloned with a new label.
     self.applied_labels = set()
+    # A boolean value indicating whether the pipeline is created in an
+    # interactive environment such as interactive notebooks. Initialized as
+    # None. The value is set ad hoc when `pipeline.run()` is invoked.
+    self.interactive = None
```

Comment:
I believe one more suggestion from the previous PR was to not keepting track of interactivity as part of the Pipeline but making it a property of the runner.

Reply:
Thanks! I'll go with your suggestion below to deduce if the job is started from a notebook environment instead of determining if the job is started from a pipeline that was originally bundled with an Interactive Runner.

Reply:
I'll go with the check environment route and make it a standalone utility module in the interactive package.